lib/encoding/EncodeData.ml

```diff
@@ -1,44 +1,81 @@
 open Ast
 open TAst
+open EncodedFile
+open Labels
+open Integers
 open ErrorUtils
-open EncodeCommon
+open PositionUtils
+open SplitFile
+
+let add_section cur_pos sec_label sec_size =
+  match MemoryAddress.next cur_pos with
+  | Some i -> i
+  | None ->
+      let txt =
+        Format.sprintf
+          "The section '%s' of size %d, is too long to be in a 32bit address \
```

Comment:
Vrai question, est-ce au moins possible ? Est-ce qu'Ocaml ne plante pas avant x) ?

Reply:
D'après mes premiers tests de ce matin : Non concluant 😅 Ça demande à OCaml de parser un fichier source de ~17Go (.text, main: et que des nop).
Lors de l'exécution, pour l'instant j'ai juste réussi à faire freeze mon OS, plus de mémoire. Je réessaye avec 100Go de swap ce soir ou demain !

Reply:
Bon j'ai essayé sur les ordis de l'ENS et... Le noyau kill le process puisqu'il utilise toute la mémoire (RAM de 16Go & Swap de 8Go). Donc objectivement non, je pense pas que l'erreur soit lancée un jour (ou à moins de laisser l'assembleur tourner quelques jours sur un serveur avec 100Go de RAM).
Mais, d'un point de vue du typage, je suis doomed, obligé d'avoir soit un assert false, ou un message d'erreur :/

Reply:
Noté ! Va pour l'erreur alors. Merci de ta motivation ;)